airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -301,6 +305,12 @@ def upload_metadata_to_gcs(bucket_name: str, metadata_file_path: Path, validator
     if metadata is None:
         raise ValueError(f"Metadata file {metadata_file_path} is invalid for uploading: {error}")
 
+    is_pre_release = validator_opts.prerelease_tag is not None
+    is_release_candidate = metadata.data.release.is_release_candidate
```

Comment:
Is it worth adding an assertion that `is_pre_release` and `is_release_candidate` aren't both `True`?

Reply:
Both can be `True`. It's determined as a pre-release from options passed to the metadata upload CLI (which originally come from an `airbyte-ci connectors publish` inputs.
So nothing prevents you from having `is_release_candidate` set to `True`, but wanting to release a `pre-release` for testing (usually from your branch).
This is why I added `should_upload_release_candidate` and `should_upload_latest` which would both be falsy if the it's a `pre-release`.